server/controllers/Followers.js

```diff
@@ -0,0 +1,151 @@
+import models from '../database/models';
+import { serverResponse, serverError } from '../helpers';
+
+const { User, UserFollower } = models;
+
+/**
+ * @export
+ * @class Followers
+ */
+class Followers {
+  /**
+   * @name follow
+   * @async
+   * @static
+   * @memberof Users
+   * @param {Object} req express request object
+   * @param {Object} res express response object
+   * @returns {JSON} JSON object with details of new follower
+   */
+  static async follow(req, res) {
+    try {
+      const {
+        user: { id },
+        params: { username }
+      } = req;
+      const user = await User.findByUsername(username);
+      if (!user) {
```

Comment:
@henryade don't you think we should move these checks from lines 27-34 to another method inside this controller and use it here and also in the unfollow route handler, cos it looks like we are repeating code. In the future we could move to a helper, if we discover that someone else is trying to do this same thing.

Reply:
Okay. I will do this now.